controllers/azurecluster_controller.go

```diff
@@ -97,7 +99,7 @@ func (acr *AzureClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 	// Add a watch on clusterv1.Cluster object for unpause notifications.
 	if err = c.Watch(
 		&source.Kind{Type: &clusterv1.Cluster{}},
-		handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("AzureCluster"))),
+		handler.EnqueueRequestsFromMapFunc(clusterToInfrastructureMapFunc(ctx, log, mgr.GetClient())),
```

Comment:
Any specific reason to pass `log` as an arg? I think you can get an instance of logger from ctx itself.
Ref: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azurecluster_controller.go#L76

Reply:
No specific reason, updated, lmk what you think